src/components/flex/Flex.jsx

```diff
@@ -0,0 +1,265 @@
+   * @see alignSelf=flex-start https://styleguide.brainly.com/latest/docs/interactive.html?alignSelf=flex-start#flexbox
+   * @see alignSelf=flex-end https://styleguide.brainly.com/latest/docs/interactive.html?alignSelf=flex-end#flexbox
+   * @see alignSelf=baseline https://styleguide.brainly.com/latest/docs/interactive.html?alignSelf=baseline#flexbox
+   * @see alignSelf=stretch https://styleguide.brainly.com/latest/docs/interactive.html?alignSelf=stretch#flexbox
+   */
+  alignSelf?: ?FlexAlignmentValuesType,
+  /**
+    * It will set flex display to inline-flex
+    * @example <Flex inlineFlex>
+    *            component content
+    *          </Flex>
+   */
+  inlineFlex?: boolean,
+  /**
+    * It will wrap component
+    * @example <Flex wrap>
+    *            component content
+    *          </Flex>
+   */
+  wrap?: boolean,
+  /**
+    * It will wrap reverse component
+    * @example <Flex wrapReverse>
+    *            component content
+    *          </Flex>
+   */
+  wrapReverse?: boolean,
+  /**
+   * Specify margin for flex based on spacings: xxs: 4px, xs: 8px, s: 16px, m: 24px, l: 40px, xl: 64px, xxl: 104px, xxxl: 168px, xxxxl: 272px
+   * @example <Flex margin="m">
+   *            component content
+   *          </Flex>
+   */
+  margin?: ?FlexMarginsType,
+  /**
+   * Specify margin top for flex based on spacings: xxs: 4px, xs: 8px, s: 16px, m: 24px, l: 40px, xl: 64px, xxl: 104px, xxxl: 168px, xxxxl: 272px
+   * @example <Flex marginTop="m">
+   *            component content
+   *          </Flex>
+   */
+  marginTop?: ?FlexMarginsType,
+  /**
+   * Specify margin right for flex based on spacings: xxs: 4px, xs: 8px, s: 16px, m: 24px, l: 40px, xl: 64px, xxl: 104px, xxxl: 168px, xxxxl: 272px
+   * @example <Flex marginRight="m">
+   *            component content
+   *          </Flex>
+   */
+  marginRight?: ?FlexMarginsType,
+  /**
+   * Specify margin bottom for flex based on spacings: xxs: 4px, xs: 8px, s: 16px, m: 24px, l: 40px, xl: 64px, xxl: 104px, xxxl: 168px, xxxxl: 272px
+   * @example <Flex marginBottom="m">
+   *            component content
+   *          </Flex>
+   */
+  marginBottom?: ?FlexMarginsType,
+  /**
+   * Specify margin left for flex based on spacings: xxs: 4px, xs: 8px, s: 16px, m: 24px, l: 40px, xl: 64px, xxl: 104px, xxxl: 168px, xxxxl: 272px
+   * @example <Flex marginLeft="m">
+   *            component content
+   *          </Flex>
+   */
+  marginLeft?: ?FlexMarginsType
+};
+
+const Flex = (props: FlexPropsType) => {
+  const {
+    fullWidth,
+    fullHeight,
+    noShrink,
+    inlineFlex,
+    alignItems,
+    alignContent,
+    justifyContent,
+    wrap,
+    wrapReverse,
+    alignSelf,
+    direction,
+    margin,
+    marginTop,
+    marginBottom,
+    marginLeft,
+    marginRight,
+    children,
+    className,
+    ...otherProps
+  } = props;
+
+  const flexClass = classNames(
```

Comment:
maybe organize it as an object? like we did in other places

Reply:
@PinkaminaDianePie fixed